.github/workflows/tox.yml

```diff
@@ -229,3 +229,31 @@ jobs:
         uses: codecov/codecov-action@v4
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
+
+      - name: Upload coverage data
+        if: ${{ (success() || failure()) && contains(matrix.coverage, 'github') && matrix.pytest == 'true' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-data-${{ matrix.os }}-${{ matrix.toxenv }}
+          path: coverage.xml
+          if-no-files-found: ignore
+
+  coverage:
+    name: test coverage
+    needs:
+      - tox
+    if: inputs.coverage == 'github'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: coverage-data-*
+          merge-multiple: true
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - run: python -Im pip install --upgrade coverage[toml]
+      - run: python -Im coverage combine
+      - run: python -Im coverage report --format=markdown >> $GITHUB_STEP_SUMMARY
+
```

Comment:
I think it would be nice to add the ability to fail this job based on either total coverage % or diff coverage %.